parser/src/main/java/com/linkedin/avroutil1/util/ConfigurableAvroSchemaComparator.java

```diff
@@ -84,115 +121,272 @@ private static boolean equals(AvroSchema a, AvroSchema b, SchemaComparisonConfig
+                AvroSchemaDifferenceType.RECORD_DEFAULT_VALUE_MISMATCH,
+                String.format("Default value %s for field %s in %s in schemaA does not match with default value %s field %s in %s in schemaB",
+                    aField.getDefaultValue(), aField.getName(), a,
+                    bField.getDefaultValue(), bField.getName(), b));
+            differences.add(difference);
+            if (fastFail) return;
           }
         } else if (aField.hasDefaultValue() || bField.hasDefaultValue()) {
-          //means one field has a default value and the other does not
-          return false;
+          AvroSchemaDifference difference = new AvroSchemaDifference(aField.getCodeLocation(), bField.getCodeLocation(),
+              AvroSchemaDifferenceType.DEFAULT_VALUE_MISSING,
+              String.format("Either field %s in %s in schemaA or field %s in %s in schemaB has missing default value",
+                            aField.getName(), a, bField.getName(), b));
+          differences.add(difference);
+          if (fastFail) return;
         }
         if (!Objects.equals(aField.getPosition(), bField.getPosition())) {
-          return false;
+          AvroSchemaDifference difference = new AvroSchemaDifference(aField.getCodeLocation(), bField.getCodeLocation(),
+              AvroSchemaDifferenceType.RECORD_FIELD_POSITION_MISMATCH,
+              String.format("Position %d for field %s in %s in schemaA does not match with position %d field %s in %s in schemaB",
+                  aField.getPosition(), aField.getName(), a, bField.getPosition(), bField.getName(), b));
+          differences.add(difference);
+          if (fastFail) return;
         }
         if (considerJsonProps && !hasSameObjectProps(aField, bField, considerJsonStringProps, considerJsonNonStringProps)) {
-          return false;
+          AvroSchemaDifference difference = new AvroSchemaDifference(aField.getCodeLocation(), bField.getCodeLocation(),
+              AvroSchemaDifferenceType.JSON_PROPERTY_MISMATCH,
+              String.format("Json properties of %s in schemaA does not match with the json properties of %s in schemaB", a, b));
+          differences.add(difference);
+          if (fastFail) return;
         }
         if (considerAliases && !hasSameAliases(aField, bField)) {
-          return false;
+          AvroSchemaDifference difference = new AvroSchemaDifference(aField.getCodeLocation(), bField.getCodeLocation(),
+              AvroSchemaDifferenceType.ALIASES_MISMATCH,
+              String.format("Aliases for %s in schemaA does not match with the aliases for %s in schemaB", a, b));
+          differences.add(difference);
+          if (fastFail) return;
         }
       }
-      return true;
+      // If there is size mismatch between the schemas, we add the additional fields in the differences
+      for (int i = numberOfCommonFields; i < aFields.size(); i++) {
```

Comment:
Seems like we are adding the same error twice: we will never have a `RECORD_FIELD_COUNT_MISMATCH` difference without an `ADDITIONAL_FIELD` difference.

I'm thinking we should remove `RECORD_FIELD_COUNT_MISMATCH`?

Reply:
Its not the same error, `RECORD_FIELD_COUNT_MISMATCH` refers to the number of fields differed in both schemas, whereas `ADDITIONAL_FIELD` refers to the exact field name which is absent from the differing schema.